client-scripts-api.md

```diff
@@ -277,3 +277,111 @@ Payload:
   value: true|false
 }
 ```
+
+### `routeChange`
+
+This event is triggered when a change in the internal router happens.
+
+Payload:
+
+```js
+{
+  location: {
+    pathname: '/products',
+    search: '?page=0',
+    query: {
+      page: 0,
+    },
+    hash: '#',
+  },
+  action: 'PUSH',
+}
+```
+
+### `searchTermChange`
+
+Subject changed and applied a new search term. Includes also the result count for the new search.
+
+Payload:
+
+```js
+{
+  searchTerm: 'chocolate',
+  resultCount: 145,
+}
+```
+
+The property `resultCount` is optional and could be omitted if result count is not known or not available.
+
+### `resultsFilterChange`
```

Comment:
💭 could maybe be just filterChange, the event origin would describe thats a results anyway right?

Reply:
Yeah, I just thought that it might potentially conflict with other filters maybe?

Reply:
but those would have different origins wouldnt they?

Reply:
we do the same if we have multiple vids, the events are generic, just differnt origins/elements that cause them